src/aws_credentials_httpc.erl

```diff
@@ -10,20 +10,29 @@
 %% @end
 
 -module(aws_credentials_httpc).
--export([start/0, get/1, get/2]).
+-export([start/0, get/1, get/2, get/3, put/1, put/2, put/3]).
 
 -define(PROFILE, aws_credentials).
 -define(TIMEOUT, 10000). % 10 sec
 -define(CONNECT_TIMEOUT, 3000). % 3 sec
 -define(DEFAULT_TRIES, 3).
+-define(DEFAULT_HEADERS, []).
 -define(DELAY, 100). % 100 microseconds
 
+-type field() :: string().
```

Comment:
Again, am I doing too much typing and spec'ing?

Reply:
No